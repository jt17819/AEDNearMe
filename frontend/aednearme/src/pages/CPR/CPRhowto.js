import { Typography } from '@mui/material';
import { Grid, Container, Button, makeStyles } from '@material-ui/core';

import CprVid from '../../components/CprVid/CprVid'

import React, { Fragment } from 'react'

import { FormControl, Input, InputLabel, MenuItem, Paper, Select, TextField } from '@mui/material'

const useStyles = makeStyles(() => ({

  text: {
    fontSize: '15px'
  },

}));



const CPRhowto = () => {

  const classes = useStyles();

  return (
    <Fragment>

    <Container 
      style={{
        paddingBottom: "80px",
        paddingTop: "5px" 
      }}>

    <h1 className={classes.title} >The four simple steps of CPR (Cardiopulmonary Resuscitation) </h1>

        <Grid container spacing={4}>

          <Grid item md={5} >
          
            <h3 className={classes.text}>
            Step 1: If you suspect someone is having a cardiac arrest, check the person over, tilt the head back and check for breathing for 10 seconds. If the person is unresponsive, ask someone to call 999/112 and tell them to bring an AED.
            </h3>
            <p>Never leave a victim by themself</p>

            <h3 className={classes.text}>
            Step 2: Start hands only CPR, lock your hands together and place them by the person's chest.
            </h3>

            <h3 className={classes.text}>
            Step 3: With straight arms, use your body to push 5-6cm in depth at a rate of 100-120 bpm. 
            </h3>
            <p>For reference, use the beat of Staying Alive by the Bee Gees or Baby Shark</p>

            <h3 className={classes.text}>
            Step 4: Keep doing hands only CPR until a defibrillator is present and the ambulance arrives. If there are multiple people around you. 
            </h3>
            <p>Switch to other helpers every 5 rounds of 30 pushes to avoid fatigue</p>
            
            <h3>Read more here at <a id="choking-link" href="https://www.sja.org.uk/get-advice/first-aid-advice/unresponsive-casualty/how-to-do-cpr-on-an-adult/" target="_blank">St. John's Ambulance</a>
            </h3>


          </Grid>

          <Grid
            item
            md={7}
            xs={12}
            className=""
            > 
            <div>
            <CprVid />
            </div>  
          </Grid>

          
        </Grid>
      </Container>

    </Fragment>
      
  )
}

export default CPRhowto