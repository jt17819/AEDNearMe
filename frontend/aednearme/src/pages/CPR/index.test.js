import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import {default as CPRhowto} from './CPRhowto';
import axios from 'axios'


describe('CPRhowto', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });


  // Rendering

  it('renders welcome message', () => {
    render(<CPRhowto />, { wrapper: MemoryRouter })
    expect(screen.getByText('The four simple steps of CPR (Cardiopulmonary Resuscitation)')).toBeInTheDocument();
  });

//   test('list contains 2 buttons', () => {
//     render(<CPRhowto />, { wrapper: MemoryRouter });
  
//     // const listElement = screen.getAllByRole('button');
//     const listItems = screen.getAllByRole('button');
  
    
//     expect(listItems.length).toEqual(1);
//   });



  // Snapshot

  it('should take a snapshot', () => {
    render(<CPRhowto />, { wrapper: MemoryRouter })
    const { asFragment } = render(<CPRhowto />, { wrapper: MemoryRouter })
    
    expect(asFragment(<CPRhowto />)).toMatchSnapshot()
  })

  

  
})