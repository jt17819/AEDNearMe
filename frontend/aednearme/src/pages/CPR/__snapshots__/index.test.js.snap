// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CPRhowto should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root pb-5 MuiContainer-maxWidthLg"
    style="padding-bottom: 100px; padding-top: 20px;"
  >
    <h1
      class="makeStyles-title-6"
    >
      The four simple steps of CPR (Cardiopulmonary Resuscitation) 
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-5"
      >
        <h3
          class="makeStyles-text-7"
        >
          Step 1: If you suspect someone is having a cardiac arrest, check the person over, tilt the head back and check for breathing for 10 seconds. If the person is unresponsive, ask someone to call 999/112 and tell them to bring an AED.
        </h3>
        <p>
          Never leave a victim by themself
        </p>
        <h3
          class="makeStyles-text-7"
        >
          Step 2: Start hands only CPR, lock your hands together and place them by the person's chest.
        </h3>
        <h3
          class="makeStyles-text-7"
        >
          Step 3: With straight arms, use your body to push 5-6cm in depth at a rate of 100-120 bpm.
        </h3>
        <p>
          For reference, use the beat of Staying Alive by the Bee Gees or Baby Shark
        </p>
        <h3
          class="makeStyles-text-7"
        >
          Step 4: Keep doing hands only CPR until a defibrillator is present and the ambulance arrives. If there are multiple people around you.
        </h3>
        <p>
          Switch to other helpers every 5 rounds of 30 pushes to avoid fatigue
        </p>
        <h3>
          Read more here at 
          <a
            href="https://www.sja.org.uk/get-advice/first-aid-advice/unresponsive-casualty/how-to-do-cpr-on-an-adult/"
            id="choking-link"
            target="_blank"
          >
            St. John's Ambulance
          </a>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <div>
          <div>
            <div
              style="width: 100%; height: 360px;"
            >
              <div
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
